modules/terraform-zscc-asg-lambda-aws/main.tf

```diff
@@ -0,0 +1,284 @@
+    actions = [
+      "autoscaling:DescribeLifecycleHookTypes",
+      "autoscaling:DescribeLifecycleHooks",
+      "autoscaling:DescribeAutoScalingInstances",
+      "autoscaling:CompleteLifecycleAction",
+      "autoscaling:RecordLifecycleActionHeartbeat",
+      "autoscaling:SetInstanceHealth",
+      "ec2:DescribeInstanceStatus"
+    ]
+    resources = ["*"]
+  }
+}
+
+# Create lambda lifecycle policy
+resource "aws_iam_policy" "lambda_autoscale_lifecycle_policy" {
+  description = "Policy which permits lambda to send lifecycle actions"
+  name        = "${var.name_prefix}-asg-lambda-aslc-${var.resource_tag}"
+  policy      = data.aws_iam_policy_document.lambda_autoscale_lifecycle_policy_document.json
+}
+
+# Attach Lambda lifeycle policy to IAM Role
+resource "aws_iam_role_policy_attachment" "lambda_autoscale_lifecycle_attachment" {
+  policy_arn = aws_iam_policy.lambda_autoscale_lifecycle_policy.arn
+  role       = aws_iam_role.asg_lambda_iam_role.name
+}
+
+
+################################################################################
+# Define AWS Managed CloudWatch Metrics Policy
+################################################################################
+# Define lambda policy for collecting Cloudwatch metrics
+data "aws_iam_policy_document" "lambda_metrics_policy_document" {
+  version = "2012-10-17"
+  statement {
+    sid    = "LambdaAllowCloudWatchMetrics"
+    effect = "Allow"
+    actions = [
+      "cloudwatch:GetMetricStatistics",
+      "cloudwatch:ListMetrics",
+      "cloudwatch:GetMetricData",
+      "ec2:DescribeTags"
+    ]
+    resources = ["*"]
+  }
+}
+
+# Create lambda metrics policy
+resource "aws_iam_policy" "lambda_metrics_policy" {
+  description = "Policy which permits lambda to retrieve custom metrics from CloudWatch"
+  name        = "${var.name_prefix}-asg-lambda-get-metrics-${var.resource_tag}"
+  policy      = data.aws_iam_policy_document.lambda_metrics_policy_document.json
+}
+
+# Attach lambda metrics policy to IAM Role
+resource "aws_iam_role_policy_attachment" "lambda_metrics_attachment" {
+  policy_arn = aws_iam_policy.lambda_metrics_policy.arn
+  role       = aws_iam_role.asg_lambda_iam_role.name
+}
+
+
+################################################################################
+# Define AWS Managed Logs Policy
+################################################################################
+# Define lambda policy for streaming logs
+data "aws_iam_policy_document" "lambda_logs_policy_document" {
+  version = "2012-10-17"
+  statement {
+    sid    = "LambdaAllowLogStreamEvents"
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    resources = [aws_cloudwatch_log_group.asg_cloudwatch_log_group.arn]
+  }
+  statement {
+    sid    = "LambdaAllowLogGroupCreation"
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup"
+    ]
+    resources = ["arn:aws:logs::${data.aws_caller_identity.asg_caller_identity.id}:*"]
```

Comment:
In @vkrishnamurthy-zscaler 's sample, he had the resource here as "arn:aws:logs:<region>:<aws_account_id>:*

Has this changed?

Reply:
we removed this with the dedicate log group 14 days retention creation